### PR TITLE
[GraphBolt] Add hardcode testcases for `gb.preprocess_ondisk_dataset`

### PR DESCRIPTION
**Pull request by Skeleton003**
## Description

Add hardcode testcases for `gb.preprocess_ondisk_dataset`

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 1]
```diff
@@ -1191,6 +1191,345 @@ def test_OnDiskDataset_preprocess_homogeneous(edge_fmt):
...
+        num_edges = 10
+        num_classes = 1
+
+        # Generate edges.
+        edges = np.array(
+            [[0, 0, 1, 1, 2, 2, 3, 3, 4, 4], [1, 2, 2, 3, 3, 4, 4, 0, 0, 1]]
+        ).T
+        os.makedirs(os.path.join(test_dir, "edges"), exist_ok=True)
+        assert edge_fmt in ["numpy", "csv"], print(
+            "only numpy and csv are supported for edges."
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:1210]:
```suggestion
            "Only numpy and csv are supported for edges."
```

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 2]
```diff
@@ -1191,6 +1191,345 @@ def test_OnDiskDataset_preprocess_homogeneous(edge_fmt):
...
+        )
+        if edge_fmt == "csv":
+            # Wrtie into edges/edge.csv
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:1213]:
```suggestion
            # Write into edges/edge.csv
```

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 3]
```diff
@@ -1191,6 +1191,345 @@ def test_OnDiskDataset_preprocess_homogeneous(edge_fmt):
...
+            edges = pd.DataFrame(edges, columns=["src", "dst"])
+            edge_path = os.path.join("edges", "edge.csv")
+            edges.to_csv(
+                os.path.join(test_dir, edge_path),
+                index=False,
+                header=False,
+            )
+        else:
+            # Wrtie into edges/edge.npy
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:1222]:
```suggestion
            # Write into edges/edge.npy
```

**Review by frozenbugs — APPROVED** [review]

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 4]
```diff
@@ -1191,6 +1191,345 @@ def test_OnDiskDataset_preprocess_homogeneous(edge_fmt):
         fused_csc_sampling_graph = None
 
 
+@pytest.mark.parametrize("edge_fmt", ["csv", "numpy"])
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:1194]: no need to test on `edge_fmt` as this is not the target we add the hardcode testcase.

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 5]
```diff
@@ -1191,6 +1191,345 @@ def test_OnDiskDataset_preprocess_homogeneous(edge_fmt):
...
+    with tempfile.TemporaryDirectory() as test_dir:
+        # All metadata fields are specified.
+        dataset_name = "graphbolt_test"
+        num_nodes = 5
+        num_edges = 10
+        num_classes = 1
+
+        # Generate edges.
+        edges = np.array(
+            [[0, 0, 1, 1, 2, 2, 3, 3, 4, 4], [1, 2, 2, 3, 3, 4, 4, 0, 0, 1]]
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:1206]: show the graph in COO format like this for better readability. https://github.com/dmlc/dgl/blob/af870387fc4a0b9b67c24fd49e5f68e133c71273/tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py#L242-L247

Comment (Rhett-Ying) [reply tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:1206]: and paste the node/edge feat as well.

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 6]
```diff
@@ -1191,6 +1191,345 @@ def test_OnDiskDataset_preprocess_homogeneous(edge_fmt):
...
+        train_path = os.path.join("set", "train.npy")
+        np.save(os.path.join(test_dir, train_path), train_data)
+        valid_data = np.array([0, 1, 2, 3, 4])
+        valid_path = os.path.join("set", "valid.npy")
+        np.save(os.path.join(test_dir, valid_path), valid_data)
+        test_data = np.array([0, 1, 2, 3, 4])
+        test_path = os.path.join("set", "test.npy")
+        np.save(os.path.join(test_dir, test_path), test_data)
+
+        yaml_content = (
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:1252]:
```
f"""
"""
```

File: tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py [hunk 7]
```diff
@@ -1191,6 +1191,345 @@ def test_OnDiskDataset_preprocess_homogeneous(edge_fmt):
...
+        )
+        assert len(subgraph.sampled_csc.indices) <= num_samples
+
+
+def test_OnDiskDataset_preprocess_heterogeneous_hardcode(edge_fmt="numpy"):
+    """Test preprocess of OnDiskDataset."""
+    with tempfile.TemporaryDirectory() as test_dir:
+        # All metadata fields are specified.
+        dataset_name = "graphbolt_test"
+        num_nodes = {
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py:1369]: ditto

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**